addon/components/flexberry-file.js

```diff
@@ -139,249 +104,302 @@ export default FlexberryBaseComponent.extend({
-    @type Boolean
-    @readonly
+    @property class
+    @type String
+    @default 'fluid'
+    @example
+    ```handlebars
+    {{flexberry-file
+      class="compact"
+      value=model.file
+    }}
+    ```
   */
-  uploadButtonIsEnabled: Ember.computed('uploadIsInProgress', 'uploadData', function() {
-    let uploadIsInProgress = this.get('uploadIsInProgress');
-    let selectedFile = this.get('selectedFile');
+  class: 'fluid',
 
-    return !(uploadIsInProgress || Ember.isNone(selectedFile));
-  }),
+  /**
+    CSS-classes names for component's add, remove, upload, download buttons.
+    See [Semantic UI buttons classes](http://semantic-ui.com/elements/button.html).
+
+    @property buttonClass
+    @type String
+    @default ''
+    @example
+    ```handlebars
+    {{flexberry-file
+      buttonClass="red"
+      value=model.file
+    }}
+    ```
+  */
+  buttonClass: '',
 
   /**
-    Flag: indicates whether download button is visible now.
+    Path to component's settings in application configuration (JSON from ./config/environment.js).
+
+    @property appConfigSettingsPath
+    @type String
+    @default 'APP.components.flexberryFile'
+  */
+  appConfigSettingsPath: 'APP.components.flexberryFile',
+
+  /**
+    Name of the action which will be send outside after click on selected image preview.
+
+    @property viewImageAction
+    @type String
+    @default 'flexberryFileViewImageAction'
+  */
+  viewImageAction: 'flexberryFileViewImageAction',
+
+  /**
+    Value of file component (contains serialized file metadata such as fileName, fileSize, etc.).
+    It is binded to related model property, so every change to value will automatically change model property.
 
-    @property downloadButtonIsVisible
+    @property value
+    @type String
+  */
+  value: null,
+
+  /**
+    Flag: indicates whether to upload file on 'relatedModel' 'preSave' event.
+
+    @property uploadOnModelPreSave
     @type Boolean
-    @readonly
+    @default true
   */
-  downloadButtonIsVisible: Ember.computed('showDownloadButton', function() {
-    // Download button is always visible (but disabled if download is not available).
-    return this.get('showDownloadButton');
-  }),
+  uploadOnModelPreSave: undefined,
 
   /**
-    Flag: indicates whether download button is enabled now.
+    Flag: indicates whether to show preview element for images or not.
 
-    @property downloadButtonIsEnabled
+    @property showPreview
     @type Boolean
-    @readonly
+    @default false
   */
-  downloadButtonIsEnabled: Ember.computed('uploadIsInProgress', 'initialValue', function() {
-    let uploadIsInProgress = this.get('uploadIsInProgress');
-    let jsonInitialValue = this.get('jsonInitialValue');
+  showPreview: false,
 
-    return !(uploadIsInProgress || Ember.isNone(jsonInitialValue));
-  }),
+  /**
+    Flag: indicates whether to show upload button or not.
+
+    @property showUploadButton
+    @type Boolean
+    @default false
```

Comment:
Yes: https://github.com/Flexberry/ember-flexberry/blob/code-formatting-and-fix-yuidoc-%233/addon/components/flexberry-file.js#L518
And we had already discussed it in one of the previous pull-requests.